tests/gtests/internals/test_utils.hpp

```diff
@@ -97,14 +97,19 @@ void write_to_dnnl_memory(const T *handle, dnnl::memory &mem, dnnl::engine &eng,
                     (const unsigned *)handle, mem_u32_mem, eng, s);
             dnnl::reorder(mem_u32_mem, mem).execute(s, mem_u32_mem, mem);
         } else {
+            s.wait();
 #ifdef DNNL_WITH_SYCL
             void *mapped_ptr = (uint8_t *)mem.get_data_handle();
+            if (!mapped_ptr)
```

Comment:
Is there a specific reason to have separate handling with SYCL?
map -> memcpy -> unmap should work with SYCL memory too, also `get_data_handle()` here won't work when `mem` uses `sycl::buffer` and not USM.

Reply:
On the OpenCL runtime, the map operation will map the entire buffer to the returned pointer. In the SYCL runtime we need to index through each handle and by default we will only get the first bytes from the buffer. I am sure there is a way to handle it but it seemed easier to perform a sycl::memcpy for this operation than create something more elaborate to handle multiple handles and make sure we transfer the correct range to the device.

Its something I can change if you prefer to not have multiple paths for the two runtimes.

Reply:
Sorry, I'm not sure I understand, To me it seems these two versions are functionally equal for SYCL USM buffers:

Version 1:

```c++
void *mapped_ptr = (uint8_t *)mem.get_data_handle();
sycl_queue.memcpy(mapped_ptr, handle, size).wait();
```

Version 2:

```c++
void *mapped_ptr = mem.map_data();
std::memcpy(mapped_ptr, handle, size);
mem.unmap_data(mapped_ptr);
```

Are they different? What is the difference?

Reply:
Looks like SYCL runtime is working fine with the map/unmap branch. I have removed that code from the branch and tests are passing.